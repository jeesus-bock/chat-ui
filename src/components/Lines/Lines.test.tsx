import { describe, expect, test } from 'vitest';
import { render, findByTestId, screen } from 'solid-testing-library';
import { Lines } from './index';

const testMsgs = [
  { type: 'msg', from: 'mock_user', to: 'mock_room', msg: 'This is a test message!', ts: 165789872275 },
  { type: 'msg', from: 'mock_user2', to: 'mock_room', msg: 'This is a test message2!', ts: 165799827275 },
];
describe('<Lines />', () => {
  test('renders', () => {
    const { container, unmount } = render(() => <Lines msgs={testMsgs} />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  test('presents messages data', async () => {
    const { queryAllByTestId, unmount } = render(() => <Lines msgs={testMsgs} />);
    const tids = queryAllByTestId('ts');
    expect(tids[0]).toHaveTextContent('10:44');
    expect(tids[1]).toHaveTextContent('1:30');
  });
});
