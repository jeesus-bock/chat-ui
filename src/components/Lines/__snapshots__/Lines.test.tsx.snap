// Vitest Snapshot v1

exports[`<Lines /> > renders 1`] = `
<div>
  <div
    class="w-full h-full overflow-y-scroll"
    data-testid="lines"
  >
    <div
      class="flex px-4 _cols_cesd7_1 grid"
    >
      <span
        class="inline-block pr-4 mr-4 border-r"
        data-testid="ts"
      >
        10:44
      </span>
      <span
        class="ml-2 pr-4 mr-4 border-r ml-0"
        data-testid="from"
      >
        <!--#-->
        mock_user
        <!--/-->
        <!--#-->
        
        <!--/-->
      </span>
      <span
        class="ml-2 ml-0 border-r"
        data-testid="msg"
      >
        This is a test message!
      </span>
    </div>
    <div
      class="flex px-4 _cols_cesd7_1 grid"
    >
      <span
        class="inline-block pr-4 mr-4 border-r"
        data-testid="ts"
      >
        01:30
      </span>
      <span
        class="ml-2 pr-4 mr-4 border-r ml-0"
        data-testid="from"
      >
        <!--#-->
        mock_user2
        <!--/-->
        <!--#-->
        
        <!--/-->
      </span>
      <span
        class="ml-2 ml-0 border-r"
        data-testid="msg"
      >
        This is a test message2!
      </span>
    </div>
  </div>
</div>
`;
